python/cuml/common/array.py

```diff
@@ -0,0 +1,314 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cupy as cp
+import numpy as np
+
+from rmm import DeviceBuffer
+from cudf.core import Buffer, Series, DataFrame
+from cuml.utils import rmm_cupy_ary
+from cuml.utils.memory_utils import _get_size_from_shape
+from cuml.utils.memory_utils import _order_to_strides
+from cuml.utils.memory_utils import _strides_to_order
+from numba import cuda
+
+
+class Array(Buffer):
+
+    """
+    Array represents an abstracted array allocation. It can be instantiated by
+    itself, creating an rmm.DeviceBuffer underneath, or can be instantiated by
+    __cuda_array_interface__ or __array_interface__ compliant arrays, in which
+    case it'll keep a reference to that data underneath. Also can be created
+    from a pointer, specifying the characteristics of the array, in that case
+    the owner of the data referred to by the pointer should be specified
+    explicitly.
+
+    To standardize our code, please import this using:
+
+    from cuml.common.array import Array as cumlArray
+
+    Parameters
+    ----------
+
+    data : rmm.DeviceBuffer, array_like, int
+        An array-like object or integer representing a
+        device or host pointer to pre-allocated memory.
+    owner : object, optional
+        Python object to which the lifetime of the memory
+        allocation is tied. If provided, a reference to this
+        object is kept in this Buffer.
+    dtype : data-type, optional
+        Any object that can be interpreted as a numpy or cupy data type.
+    shape : int or tuple of ints, optional
+        Shape of created array.
+    order: string, optional
+        Whether to create a F-major or C-major array.
+
+    Attributes
+    ----------
+
+    ptr : int
+        Pointer to the data
+    size : int
+        Size of the array data in bytes
+    _owner : Python Object
+        Object that owns the data of the array
+    shape : tuple of ints
+        Shape of the array
+    order : {'F', 'C'}
+        'F' or 'C' to indicate Fortran-major or C-major order of the array
+    strides : tuple of ints
+        Strides of the data
+    __cuda_array_interface__ : dictionary
+        __cuda_array_interface__ to interop with other libraries.
+
+    Object Methods
+    --------------
+
+    to_output : Convert the array to the appropriate output format.
+
+    Class Methods
+    -------------
+
+    Array.empty : Create an empty array, allocating a DeviceBuffer.
```

Comment:
It looks like you added the ones, and random class methods below but didn't reference them here.